### DISCUSSION TITLE
How to set the "original" tokyonight as my colorscheme?

### DISCUSSION THREAD

**Discussion by amirhhashemi** [NvChad/NvChad#892] [Q&A]
The NvChad's tokyonight theme is slightly different from the [original one](https://github.com/folke/tokyonight.nvim). It has more red!

**Original:**
![image](https://user-images.githubusercontent.com/87268103/159293045-8dadd2c3-d9f6-4c54-aec2-c99e39771270.png)

**NvChad:**
![image](https://user-images.githubusercontent.com/87268103/159293267-469915ef-1ea5-42fe-aeaa-c9a368d0439c.png)

So I installed the original tokyonight and set it as my colorscheme in the `custom/init.lua` file:
```lua
vim.cmd "colorscheme tokyonight"
```
**This is the result:**
![image](https://user-images.githubusercontent.com/87268103/159294208-8817774b-2b06-4312-a5fa-f86abba8aaa3.png)

That's closer to what I expected, but some of the colors are still different. For example, HTML tags are yellow which shouldn't be.

Is it possible that the NvChad's tokyonight is overriding some of the original tokyonight highlights? If yes how could I prevent that?

UPDATE: 
https://github.com/NvChad/NvChad/discussions/892#discussioncomment-2598085

**Comment by siduck** [comment]
We're working on making themes close to the og theme so this is gonna take a while! I'll drop the solution after my exams finish ( this month's ending )

**Reply by sillypnk** [reply]
Can't we just make like 
```lua
local colors = {
  -- ... --
TSThing = ""
}

return colors
```

**Reply by siduck** [reply]
> Can't we just make like
> 
> ```lua
> local colors = {
>   -- ... --
> TSThing = ""
> }
> 
> return colors
> ```

it'll be easier than that, so wait for some days :rocket:

**Reply by amirhhashemi** [reply]
@siduck Let me know if I can do anything practical to help. I have a lot of free time at the end of this month.

**Reply by siduck** [reply]
> @siduck Let me know if I can do anything practical to help. I have a lot of free time at the end of this month.

oh cool!! you can help with the existing issues. I'll work on the theming part after exams

**Comment by amirhhashemi** [comment]
I managed to change some of the incorrect highlights so it looks more like the original one.

You should set `tokyonight` as your NvChad theme and install [`folke/tokyonight.nvim`](https://github.com/folke/tokyonight.nvim) and set it as your theme in `custom/init.lua`:
```lua
vim.cmd "colorscheme tokyonight"
```
Then add `hl_override` to your `chadrc.lua`. `custom.highlights` is the path of the file that is going to override highlights.
```lua
M.options = {
   hl_override = "custom.highlights",
}
```

This is the content of `custom/highlights.lua`:

```lua
local fg = require("core.utils").fg
local bg = require("core.utils").bg

local c = {
   none = "NONE",
   bg_dark = "#1f2335",
   bg = "#24283b",
   bg_highlight = "#292e42",
   terminal_black = "#414868",
   fg = "#c0caf5",
   fg_dark = "#a9b1d6",
   fg_gutter = "#3b4261",
   dark3 = "#545c7e",
   comment = "#565f89",
   dark5 = "#737aa2",
   blue0 = "#3d59a1",
   blue = "#7aa2f7",
   cyan = "#7dcfff",
   blue1 = "#2ac3de",
   blue2 = "#0db9d7",
   blue5 = "#89ddff",
   blue6 = "#B4F9F8",
   blue7 = "#394b70",
   magenta = "#bb9af7",
   magenta2 = "#ff007c",
   purple = "#9d7cd8",
   orange = "#ff9e64",
   yellow = "#e0af68",
   green = "#9ece6a",
   green1 = "#73daca",
   green2 = "#41a6b5",
   teal = "#1abc9c",
   red = "#f7768e",
   red1 = "#db4b4b",
   git = { change = "#6183bb", add = "#449dab", delete = "#914c54", conflict = "#bb7a61" },
   gitSigns = { add = "#164846", change = "#394b70", delete = "#823c41" },
}

vim.cmd "hi Statement guifg=9d7cd8 gui=none"

fg("TSProperty", c.green1)
fg("TSTag", c.magenta)
fg("TSTagDelimiter", c.blue2)
fg("TSType", c.blue1)
fg("TSOperator", c.blue5)
fg("TSParameter", c.yellow)
fg("Label", c.cyan)
fg("TSKeyword", c.purple)
fg("TSConditional", c.magenta)
fg("TSInclude", c.cyan)
fg("TSConstructor", c.magenta)
bg("Normal", c.bg)
bg("CursorLine", c.bg_highlight)
```
